### PR TITLE
HLM-728 Global interceptor and toast

### PR DESCRIPTION
**Pull request by vishwajeet-eGov**
@saiprakash-v @prasanna-egov . Please review.

### CODE REVIEW

File: tenant-portal/package.json [hunk 1]
```diff
@@ -16,6 +16,7 @@
     "react-dom": "^18.2.0",
     "react-router-dom": "^6.3.0",
     "react-scripts": "4.0.1",
+    "react-toastify": "^8.0.0",
```
Comment (prasanna-egov) [inline tenant-portal/package.json:19]: Does bootstrap not have a toast plugin?

File: tenant-portal/src/App.js [hunk 2]
```diff
@@ -7,6 +7,23 @@ import { PrivateRoute } from "./utils/PrivateRoute";
 import CreateSchema from "./components/CreateSchema/CreateSchema";
 import config from "./config.json"
 import Footer from "./components/Footer/Footer";
+import { ToastContainer, toast } from 'react-toastify';
+import 'react-toastify/dist/ReactToastify.css';
```
Comment (prasanna-egov) [inline tenant-portal/src/App.js:11]: Do we need to import the CSS, shouldn't toast plugin import it within itself.

File: tenant-portal/src/App.js [hunk 3]
```diff
@@ -7,6 +7,23 @@ import { PrivateRoute } from "./utils/PrivateRoute";
...
+import axios from 'axios';
+
+axios.interceptors.response.use(
+  response => response,
+  error => {
+    if (error.response.status === 401) {
+      toast.error("Incorrect User name/Password");
```
Comment (prasanna-egov) [inline tenant-portal/src/App.js:18]: This is wrong. Username/password auth happens in keycloak. Here we will redirect to login screen. react-keycloak has a function we can leverage for this.

File: tenant-portal/src/App.js [hunk 4]
```diff
@@ -7,6 +7,23 @@ import { PrivateRoute } from "./utils/PrivateRoute";
...
+    }
+    else if (error.response.status === 403) {
+      toast.error("Error 403");
```
Comment (prasanna-egov) [inline tenant-portal/src/App.js:21]: The message will be You are not authorized to view this resource.

File: tenant-portal/src/App.js [hunk 5]
```diff
@@ -7,6 +7,23 @@ import { PrivateRoute } from "./utils/PrivateRoute";
...
+    }
+    else if(error.response){
+      toast.error("Some error occured.");
```
Comment (prasanna-egov) [inline tenant-portal/src/App.js:24]: Please display the error message that comes from the backend in this case.

**Review by prasanna-egov — APPROVED** [review]

File: tenant-portal/package.json (hunk 1 above)
Comment (vishwajeet-eGov) [reply tenant-portal/package.json:19]: Yes, bs also has toast plugin.

File: tenant-portal/src/App.js (hunk 2 above)
Comment (vishwajeet-eGov) [reply tenant-portal/src/App.js:11]: Yes, It needed an import.

File: tenant-portal/package.json (hunk 1 above)
Comment (prasanna-egov) [reply tenant-portal/package.json:19]: can you use that ?

**Comment by vishwajeet-eGov** [comment]
Updated the code to use bootstrap toast and opened a new PR.

**Pull request closed without merging**